src/core/toga/widgets/table.py

```diff
@@ -136,3 +140,62 @@ def on_select(self, handler):
         """
         self._on_select = wrapped_handler(self, handler)
         self._impl.set_on_select(self._on_select)
+
+    def add_column(self, heading, accessor=None):
+        """
+        Add a new column to the table
+
+        :param heading:     title of the column
+        :type heading:      ``string``
+        :param accessor:    accessor of this new column
+        :type heading:      ``string``
+        """
+
+        if not accessor:
+            accessor = to_accessor(heading)
+
+        if accessor in self._accessors:
+            raise ValueError('Accesor "{}" is already in use'.format(accessor))
+
+        self.headings.append(heading)
+        self._accessors.append(accessor)
+
+        self._impl.add_column(heading, accessor)
+
+    def remove_column(self, column):
+        """
+        Remove a table column
+
+        :param column:     accessor or position (>0)
+        :type column:      ``string``
+        :type column:      ``int``
+        """
+
+        if type(column) == str and column in self._accessors:
+            # delete by accessor
+            accessor = column
+        elif type(column) == int and \
+                column >= 0 and column <= len(self._accessors)-1:
```

Comment:
This value checking is unnecessary; you can wrap the array access in a try/catch block, and catch the IndexError. That will (a) be faster for good lookups, and (b) catch both possible failure cases. It would also allow the `type(column) == int` check to be removed, as any attempt to use a non-integer will raise a TypeError.

Reply:
Fixed